arelle/plugin/validate/EDINET/rules/gfm.py

```diff
@@ -1023,6 +1023,97 @@ def rule_gfm_1_7_1(
+    EDINET.EC5700W: [GFM 1.7.2] All calculation relationships must have a weight of either 1 or -1
+    """
+    calculationRelationshipSet = val.modelXbrl.relationshipSet(tuple(LinkbaseType.CALCULATION.getArcroles()))
+    if calculationRelationshipSet is None:
+        return
+    for rel in calculationRelationshipSet.modelRelationships:
+        if rel.weight not in [1, -1]:
+            yield Validation.warning(
+                codes='EDINET.EC5700W.GFM.1.7.2',
+                msg=_("The calculation relationship must have a weight of 1 or -1, actual weight: '%(weight)s'"),
+                weight=rel.weight,
+                modelObject=rel
+            )
+
+
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+    disclosureSystems=[DISCLOSURE_SYSTEM_EDINET],
+)
+def rule_gfm_1_7_3(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    EDINET.EC5700W: [GFM 1.7.3] The concepts participating in a calculation relationship must have the same period type
+    """
+    calculationRelationshipSet = val.modelXbrl.relationshipSet(tuple(LinkbaseType.CALCULATION.getArcroles()))
+    if calculationRelationshipSet is None:
+        return
+    for rel in calculationRelationshipSet.modelRelationships:
+        fromConcept = rel.fromModelObject
+        toConcept = rel.toModelObject
+        if fromConcept.periodType != toConcept.periodType:
+            yield Validation.warning(
+                codes='EDINET.EC5700W.GFM.1.7.3',
+                msg=_("The concepts participating in a calculation relationship must have the same period types. "
+                      "The concept of '%(concept1)s' has a period type of '%(concept1PeriodType)s' and the concept "
+                      "of '%(concept2)s' has a period type of '%(concept2PeriodType)s'"),
+                concept1=fromConcept.qname,
+                concept1PeriodType=fromConcept.periodType,
+                concept2=toConcept.qname,
+                concept2PeriodType=toConcept.periodType,
+                modelObject=rel
+            )
+
+
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+    disclosureSystems=[DISCLOSURE_SYSTEM_EDINET],
+)
+def rule_gfm_1_7_6(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    EDINET.EC5700W: [GFM 1.7.6] Calculation relationships must have unique order attributes
+    """
+    calculationRelationshipSet = val.modelXbrl.relationshipSet(tuple(LinkbaseType.CALCULATION.getArcroles()))
+    if calculationRelationshipSet is None:
+        return
+    for modelObject, rels in calculationRelationshipSet.loadModelRelationshipsFrom().items():
+        if len(rels) <= 1:
+            continue
+        relsByOrder = defaultdict(list)
+        for rel in rels:
+            relsByOrder[(rel.arcElement.get("order"), rel.linkrole)].append(rel)
```

Comment:
Should this rule take into account default order values (`1.0` if not set)? `rel.order` handles this automatically.

Reply:
The rule specifically calls out that Presentation, Calculation and Definition relationships have to have an `order` attribute. If Arelle defaults to `1.0` that hides the fact that the attribute is not present in the XML.

Reply:
For this rule, it makes sense not to use the `1.0` default, but maybe we should not add elements without an `order` attribute here since it will be flagged by the other rule. As-is, two arcs without an `order` attribute would be flagged as duplicates here, which seems redundant with the other rule.